xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/XDOMBuilder.java

```diff
@@ -52,13 +52,14 @@ public XDOMBuilder()
      */
     public XDOM getXDOM()
     {
-        List<Block> blocks = endBlockList();
-
+        
         if (!this.stack.isEmpty()) {
             throw new IllegalStateException("Unbalanced begin/end Block events, missing " + this.stack.size()
                 + " calls to endBlockList().");
         }
-
+        
+        List<Block> blocks = endBlockList();
```

Comment:
hmm this is not going to have the same effect since `endBlockList()` calls `this.stack.pop()` , and thus `this.stack.isEmpty()` is going to have a different result value.

Reply:
So would this be a false positive or is there another way to fix this?

Reply:
I think I found a solution can you check it?